addons/mail/models/mail_message.py

```diff
@@ -876,10 +876,7 @@ def _message_format(self, fnames, format_reply=True):
                 thread_ids_by_model_name[message.model].add(message.res_id)
         for vals in vals_list:
             message_sudo = self.browse(vals['id']).sudo().with_prefetch(self.ids)
-            author = {
-                'id': message_sudo.author_id.id,
-                'name': message_sudo.author_id.name,
-            } if message_sudo.author_id else [('clear',)]
+            author = message_sudo.author_id.mail_partner_format({'id': True, 'name': True, 'user': {"id": True}}).get(message_sudo.author_id) if message_sudo.author_id else [('clear',)]
```

Comment:
note: we have a coming conflicting PR/fix that will add `is_company` here as well

Reply:
FYI, that's the commit in question with `is_company`: https://github.com/odoo/odoo/pull/133311 (merged imminently)